core/domain/skill_domain.py

```diff
@@ -92,7 +92,7 @@ def __init__(self, change_dict):
         self.cmd = change_dict['cmd']
 
         if self.cmd == CMD_ADD_SKILL_MISCONCEPTION:
-            self.misconception_id = change_dict['id']
+            self.new_value = change_dict['new_value']
```

Comment:
Why was this changed? or if you are passing a new misconception dict when creating it, maybe have it misconception_dict?

Reply:
I thought it seemed a bit network-inefficient to create the misconception with the id, wait for the response, and then update the misconception's values every time a new misconception is created. A misconception with just an id isn't ever created.

Reply:
Ok, yeah, that's fine. So, then renaming incoming field to new_misconception_dict is enough (which is now done).